src/Graphics/Image/Processing/Deconvolution.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE BangPatterns #-} 
+
+module Graphics.Image.Processing.Deconvolution where
+
+import Graphics.Image as I
+import Graphics.Image.Interface as I
+import Prelude as P
+
+-- | Supplementary function needed to blur an image and test the algorithm.
+randomBlur :: Image VS X Double
+randomBlur = randomBlur' / scalar (I.sum randomBlur')
+  where randomBlur' = fromLists [[0, 0, 4, 3, 2]
+                                ,[0, 1, 3, 4, 3]
+                                ,[1, 2, 3, 3, 4]
+                                ,[0, 1, 2, 1, 0]
+                                ,[0, 0, 1, 0, 0]]
+
+-- | 'deconvolve' is used to correct the systematic error of blur (loss of 
+-- contrast in smaller features) in images. This particular variant is used
+-- when no information about the distortion (blurring and noise) is known.
+-- More info about deconvolution at <https://en.wikipedia.org/wiki/Deconvolution>
+-- 
+-- <<images/frog.jpg>>   <<images/frog_deconvolve.jpg>>
+-- 
+--  Usage:
+-- >>> u0 <- readImage' "images/frog.jpg" :: IO (Image VS RGB Double)
+-- >>> let p = randomBlur
+-- >>> let d = convolve Edge p u0
+-- >>> let us = iterate (deconvolve p d) d  -- Iterative deconvolution
+-- >>>     u1 = us !! 1 -- one iteration
+-- >>>     u2 = us !! 20 -- twenty iterations
+-- >>> let output = makeImage (rows u0, cols u0 * 4)
+-- >>>        (\(r,c) ->
+-- >>>           let (i, c') = c `quotRem` cols u0
+-- >>>           in index ([u0,d,u1,u2] !! i) (r,c'))
+-- >>>        :: Image VS RGB Double
+-- >>> writeImage "images/frog_deconvolve.jpg" output
+-- 
+deconvolve :: Image VS X Double
+           -> Image VS RGB Double
+           -> Image VS RGB Double
+           -> Image VS RGB Double
+deconvolve p d u
```

Comment:
This algorithm isn't restricted to RGB images only. Also restricting to `Edge` might be a bad idea, so let's have it available as another argument.

Reply:
@lehins Okay sir, just one thing. I'll try my best to address the feedback on all my PR's asap but I've lectures and lab sessions today so won't be able to work on them before night time. Hope you understand. Thank you!

Reply:
@lehins Sorry sir, I didn't exactly understand. Do you mean that I should take input images as `Image arr Y` ?

Reply:
No, what I meant is, it should be more polymorphic. `deconvolve` is the opposite of `convolve`, so look at it's type signature and you'll see there is no reason it shouldn't be almost the same:
```
convolve :: (Array arr X e, Array arr cs e) =>
            Border (Pixel cs e) -- ^ Approach to be used near the borders.
         -> Image arr X e -- ^ Kernel image.
         -> Image arr cs e -- ^ Source image.
         -> Image arr cs e
```

So, try to fix up the code so the type signature is:

```
deconvolve :: (Array arr X e, Array arr cs e) =>
            Border (Pixel cs e) -- ^ Approach to be used near the borders.
         -> Image arr X e -- ^ Blurring Kernel.
         -> Image arr cs e -- ^ Image that is being deconvolved.
         -> Image arr cs e -- ^ Image from previous iteration of deconvolution
         -> Image arr cs e -- ^ Deblurred image
```